VCFv4.3.tex

```diff
@@ -503,41 +503,100 @@ \subsection{VCF tag naming conventions}
 
 \section{INFO keys used for structural variants}
 \label{sv-info-keys}
-The following INFO keys are reserved for encoding structural variants. In general, when these keys are used by imprecise variants, the values should be best estimates. When a key reflects a property of a single alt allele (e.g. SVLEN), then when there are multiple alt alleles there will be multiple values for the key corresponding to each alelle (e.g. SVLEN=-100,-110 for a deletion with two distinct alt alleles).
+The following INFO keys are reserved for encoding structural variants. When a key reflects a property of a single alt allele (e.g. SVLEN), then when there are multiple alt alleles there will be multiple values for the key corresponding to each alelle (e.g. SVLEN=-100,-110 for a deletion with two distinct alt alleles).
 \footnotesize
 \begin{verbatim}
 ##INFO=<ID=IMPRECISE,Number=0,Type=Flag,Description="Imprecise structural variation">
 ##INFO=<ID=NOVEL,Number=0,Type=Flag,Description="Indicates a novel structural variation">
 ##INFO=<ID=END,Number=1,Type=Integer,Description="End position of the variant described in this record">
 \end{verbatim}
 \normalsize
-For precise variants, END is POS + length of REF allele - 1, and the for imprecise variants the corresponding best estimate.
+For precise variants, END is POS + length of REF allele - 1.
 
 \footnotesize
 \begin{verbatim}
 ##INFO=<ID=SVTYPE,Number=1,Type=String,Description="Type of structural variant">
 \end{verbatim}
 \normalsize
-This key can be derived from the REF/ALT fields but is useful for filtering. The reserved values must be used for the types listed below:
+SVTYPE is required for all structural variants. The following reserved values should be used whenever possible; indented values are subtypes that should be used if evidence supports them (see examples in Section 5.3, Encoding Structural Variants):\vspace{0.3cm}\\
 \begin{itemize}
-  \item DEL: Deletion relative to the reference
-  \item INS: Insertion of novel sequence relative to the reference
-  \item DUP: Region of elevated copy number relative to the reference
-  \item INV: Inversion of reference sequence
-  \item CNV: Copy number variable region (may be both deletion and duplication)
+\item [\null] INS:  Insertion relative to the reference
+	\begin{itemize}
+	\item [\null] MEI:  Mobile element insertion relative to the reference
+		\begin{itemize}
+		\item [\null] ALU:  Alu mobile element insertion relative to the reference
+		\item [\null] LINE1:  LINE1 mobile element insertion relative to the reference
+		\item [\null] SVA:  SVA mobile element insertion relative to the reference
+		\item [\null] HERV:  HERV mobile element insertion relative to the reference
+		\end{itemize}
+	\item [\null] RET:  Retrotransposon insertion relative to the reference
+	\end{itemize}
+\item [\null] DEL:  Deletion relative to the reference
+	\begin{itemize}
+	\item [\null] -MEI:  Absence of a mobile element present in the reference
```

Comment:
I would prefer DEL:MEI over DEL:-MEI

Reply:
The use of colons to construct nested strings of types and subtypes is only described in the spec for the ALT field (symbolic ALT alleles). I have never seen this notation used in the SVTYPE field. Have you?

My intention with the indented list of subtypes in this PR is that only one value for SVTYPE should be used; the granularity of the choice of term depends on the evidence present in the data. For example, if an insertion is observed but nothing definitive is known (yet) about its content, SVTYPE=INS. However, if the same insertion is known to be a mobile element of some kind (family not specified), then SVTYPE=MEI. If it is known that the insertion is a LINE1 element, then SVTYPE=LINE1 (perhaps L1 would be better).

Thus the nesting would be implied. I proposed this schema mainly because it is consistent with the seemingly widely adopted practice (which ignores guidance provided by the current spec), of indicating mobile element insertions by using SVTYPE values such as ALU, LINE1, and SVA.

In keeping with this "one-word SVTYPE" controlled vocabulary, I proposed "-ALU", for example, to represent a so-called Alu-element deletion (more accurately described as the absence of an Alu insertion which is present in the reference genome). For an up-to-date treatment of mobile element variation, please see https://www.ncbi.nlm.nih.gov/pubmed/28855259.

I wanted to make the above points regardless of the mechanism that is eventually chosen to identify event types (like your proposed SVCLASS).

Reply:
> The use of colons to construct nested strings of types and subtypes is only described in the spec for the ALT field (symbolic ALT alleles). I have never seen this notation used in the SVTYPE field. Have you?

Both Pindel and manta use `<DUP:TANDEM>` alt alleles, with pindel using `SVTYPE=DUP:TANDEM` notation (manta just calls `SVTYPE=DUP`).

Reply:
@d-cameron is that the only example you are aware of, or was it intended as one example that is part of an argument that colons _should_ be used in SVTYPE?

`ALT=<DUP:TANDEM>` adheres to the existing spec's guidelines for the use of symbolic alternate alleles. Pindel's use of `SVTYPE=DUP:TANDEM` does not, and it seems like an effort to address exactly the sort of deficiency in the spec that I aim to improve with this PR. Even though `SVTYPE=DUP` is "legal" and `SVTYPE=DUP:TANDEM` is not, IMO the latter is better in that it imparts crucial information about the variant which the former does not.

Reply:
@thefferon the example was just a comment on current usage. I'm of the opinion that colons should be used if the values are an open set, but are not necessary if the set is closed (since the hierarchy can be defined in the spec)

Reply:
Keeping the parent (hierarchical) information in the ALT helps grep-ing